### ISSUE TITLE
cant install

### ISSUE THREAD

**Issue by NaturesProphet** [EricSmekens/node-bluetooth-obd#36]
when i do 

```
npm install bluetooth-obd
```

it try to compile something and throws error.

**Comment by Tobstr02** [comment]
Sam issue here, i'll try to give more description to the issue.

When i try to install it using `npm install bluetooth-obd`, i receive an error while the installer runs node-gyp.

I'm using node version v10.19.0 and npm version 6.14.4.
Maybe the issue here is that some libraries where the bluetooth-obd library is built on changed etc.


```
0 verbose cli [
0 verbose cli   '/usr/local/bin/node',
0 verbose cli   '/usr/local/bin/npm',
0 verbose cli   'install',
0 verbose cli   'bluetooth-obd'
0 verbose cli ]
1 info using npm@7.21.0
2 info using node@v14.17.5
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 1ms
6 timing config:load:builtin Completed in 1ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:/tmp/obd2-test/.npmrc Completed in 0ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:/home/xx/.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 5ms
19 timing npm:load:configload Completed in 5ms
20 timing npm:load:setTitle Completed in 0ms
21 timing npm:load:setupLog Completed in 1ms
22 timing config:load:flatten Completed in 1ms
23 timing npm:load:cleanupLog Completed in 2ms
24 timing npm:load:configScope Completed in 0ms
25 timing npm:load:projectScope Completed in 0ms
26 timing npm:load Completed in 9ms
27 timing arborist:ctor Completed in 0ms
28 timing idealTree:init Completed in 10ms
29 timing idealTree:userRequests Completed in 2ms
30 silly idealTree buildDeps
31 silly fetch manifest bluetooth-obd@*
32 verbose shrinkwrap failed to load node_modules/.package-lock.json out of date, updated: node_modules
33 http fetch GET 200 https://registry.npmjs.org/bluetooth-obd 129ms (cache revalidated)
34 silly placeDep ROOT bluetooth-obd@0.2.4 OK for: obd2-test@1.0.0 want: *
35 silly fetch manifest bluetooth-serial-port@~2.1.0
36 http fetch GET 200 https://registry.npmjs.org/bluetooth-serial-port 808ms (cache revalidated)
37 timing idealTree:#root Completed in 958ms
38 silly placeDep ROOT bluetooth-serial-port@2.1.7 OK for: bluetooth-obd@0.2.4 want: ~2.1.0
39 silly fetch manifest bindings@1.2.x
40 silly fetch manifest nan@latest
41 silly fetch manifest @types/node@^7.0.10
42 http fetch GET 200 https://registry.npmjs.org/bindings 66ms (cache revalidated)
43 http fetch GET 200 https://registry.npmjs.org/@types%2fnode 127ms (cache revalidated)
44 http fetch GET 200 https://registry.npmjs.org/nan 634ms (cache revalidated)
45 timing idealTree:node_modules/bluetooth-obd Completed in 647ms
46 silly placeDep ROOT @types/node@7.10.14 OK for: bluetooth-serial-port@2.1.7 want: ^7.0.10
47 silly placeDep ROOT bindings@1.2.1 OK for: bluetooth-serial-port@2.1.7 want: 1.2.x
48 silly placeDep ROOT nan@2.15.0 OK for: bluetooth-serial-port@2.1.7 want: latest
49 timing idealTree:node_modules/bluetooth-serial-port Completed in 25ms
50 timing idealTree:node_modules/@types/node Completed in 0ms
51 timing idealTree:node_modules/bindings Completed in 0ms
52 timing idealTree:node_modules/nan Completed in 0ms
53 timing idealTree:buildDeps Completed in 1634ms
54 timing idealTree:fixDepFlags Completed in 2ms
55 timing idealTree Completed in 1649ms
56 timing reify:loadTrees Completed in 1650ms
57 timing reify:diffTrees Completed in 1ms
58 silly reify mark retired [
58 silly reify   '/tmp/obd2-test/node_modules/nopt',
58 silly reify   '/tmp/obd2-test/node_modules/.bin/nopt'
58 silly reify ]
59 silly reify mark retired [ '/tmp/obd2-test/node_modules/abbrev' ]
60 silly reify moves {
60 silly reify   '/tmp/obd2-test/node_modules/nopt': '/tmp/obd2-test/node_modules/.nopt-c3SrP9AO',
60 silly reify   '/tmp/obd2-test/node_modules/.bin/nopt': '/tmp/obd2-test/node_modules/.bin/.nopt-bKYJ42rr',
60 silly reify   '/tmp/obd2-test/node_modules/abbrev': '/tmp/obd2-test/node_modules/.abbrev-IJH3RU2e'
60 silly reify }
61 timing reify:retireShallow Completed in 4ms
62 timing reify:createSparse Completed in 1ms
63 timing reify:loadBundles Completed in 0ms
64 silly audit bulk request {
64 silly audit   'bluetooth-obd': [ '0.2.4' ],
64 silly audit   'bluetooth-serial-port': [ '2.1.7' ],
64 silly audit   '@types/node': [ '7.10.14' ],
64 silly audit   bindings: [ '1.2.1' ],
64 silly audit   nan: [ '2.15.0' ]
64 silly audit }
65 timing reifyNode:node_modules/bindings Completed in 32ms
66 timing reifyNode:node_modules/@types/node Completed in 34ms
67 timing reifyNode:node_modules/bluetooth-obd Completed in 34ms
68 timing reifyNode:node_modules/bluetooth-serial-port Completed in 42ms
69 timing reifyNode:node_modules/nan Completed in 45ms
70 timing reify:unpack Completed in 45ms
71 timing reify:unretire Completed in 0ms
72 timing build:queue Completed in 1ms
73 info run bluetooth-serial-port@2.1.7 install node_modules/bluetooth-serial-port node-gyp configure build
74 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/advisories/bulk 226ms
75 timing auditReport:getReport Completed in 228ms
76 silly audit report {}
77 timing auditReport:init Completed in 0ms
78 timing reify:audit Completed in 228ms
79 info run bluetooth-serial-port@2.1.7 install { code: 1, signal: null }
80 timing reify:rollback:createSparse Completed in 23ms
81 timing reify:rollback:retireShallow Completed in 1ms
82 timing command:install Completed in 2485ms
83 verbose stack Error: command failed
83 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
83 verbose stack     at ChildProcess.emit (events.js:400:28)
83 verbose stack     at maybeClose (internal/child_process.js:1055:16)
83 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
84 verbose pkgid bluetooth-serial-port@2.1.7
85 verbose cwd /tmp/obd2-test
86 verbose Linux 5.11.0-37-generic
87 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "bluetooth-obd"
88 verbose node v14.17.5
89 verbose npm  v7.21.0
90 error code 1
91 error path /tmp/obd2-test/node_modules/bluetooth-serial-port
92 error command failed
93 error command sh -c node-gyp configure build
94 error make: Verzeichnis „/tmp/obd2-test/node_modules/bluetooth-serial-port/build“ wird betreten
94 error   CXX(target) Release/obj.target/BluetoothSerialPort/src/linux/BluetoothSerialPort.o
94 error make: Verzeichnis „/tmp/obd2-test/node_modules/bluetooth-serial-port/build“ wird verlassen
95 error gyp info it worked if it ends with ok
95 error gyp info using node-gyp@7.1.2
95 error gyp info using node@14.17.5 | linux | x64
95 error gyp info find Python using Python version 3.8.10 found at "/usr/bin/python3"
95 error gyp info spawn /usr/bin/python3
95 error gyp info spawn args [
95 error gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
95 error gyp info spawn args   'binding.gyp',
95 error gyp info spawn args   '-f',
95 error gyp info spawn args   'make',
95 error gyp info spawn args   '-I',
95 error gyp info spawn args   '/tmp/obd2-test/node_modules/bluetooth-serial-port/build/config.gypi',
95 error gyp info spawn args   '-I',
95 error gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
95 error gyp info spawn args   '-I',
95 error gyp info spawn args   '/home/xx/.cache/node-gyp/14.17.5/include/node/common.gypi',
95 error gyp info spawn args   '-Dlibrary=shared_library',
95 error gyp info spawn args   '-Dvisibility=default',
95 error gyp info spawn args   '-Dnode_root_dir=/home/xx/.cache/node-gyp/14.17.5',
95 error gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
95 error gyp info spawn args   '-Dnode_lib_file=/home/xx/.cache/node-gyp/14.17.5/<(target_arch)/node.lib',
95 error gyp info spawn args   '-Dmodule_root_dir=/tmp/obd2-test/node_modules/bluetooth-serial-port',
95 error gyp info spawn args   '-Dnode_engine=v8',
95 error gyp info spawn args   '--depth=.',
95 error gyp info spawn args   '--no-parallel',
95 error gyp info spawn args   '--generator-output',
95 error gyp info spawn args   'build',
95 error gyp info spawn args   '-Goutput_dir=.'
95 error gyp info spawn args ]
95 error gyp info spawn make
95 error gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
95 error In file included from ../src/linux/BluetoothSerialPort.cc:13:
95 error ../src/DeviceINQ.h:49:30: error: ‘v8::Handle’ has not been declared
95 error    49 |         static void Init(v8::Handle<v8::Object> exports);
95 error       |                              ^~~~~~
95 error ../src/DeviceINQ.h:49:36: error: expected ‘,’ or ‘...’ before ‘<’ token
95 error    49 |         static void Init(v8::Handle<v8::Object> exports);
95 error       |                                    ^
95 error In file included from ../src/linux/BluetoothSerialPort.cc:14:
95 error ../src/BTSerialPortBinding.h:43:30: error: ‘v8::Handle’ has not been declared
95 error    43 |         static void Init(v8::Handle<v8::Object> exports);
95 error       |                              ^~~~~~
95 error ../src/BTSerialPortBinding.h:43:36: error: expected ‘,’ or ‘...’ before ‘<’ token
95 error    43 |         static void Init(v8::Handle<v8::Object> exports);
95 error       |                                    ^
95 error ../src/linux/BluetoothSerialPort.cc:18:14: error: variable or field ‘InitAll’ declared void
95 error    18 | void InitAll(Handle<Object> exports) {
95 error       |              ^~~~~~
95 error ../src/linux/BluetoothSerialPort.cc:18:14: error: ‘Handle’ was not declared in this scope
95 error ../src/linux/BluetoothSerialPort.cc:18:27: error: expected primary-expression before ‘>’ token
95 error    18 | void InitAll(Handle<Object> exports) {
95 error       |                           ^
95 error ../src/linux/BluetoothSerialPort.cc:18:29: error: ‘exports’ was not declared in this scope
95 error    18 | void InitAll(Handle<Object> exports) {
95 error       |                             ^~~~~~~
95 error In file included from ../src/linux/BluetoothSerialPort.cc:12:
95 error ../src/linux/BluetoothSerialPort.cc:23:34: error: ‘InitAll’ was not declared in this scope
95 error    23 | NODE_MODULE(BluetoothSerialPort, InitAll)
95 error       |                                  ^~~~~~~
95 error /home/xx/.cache/node-gyp/14.17.5/include/node/node.h:758:36: note: in definition of macro ‘NODE_MODULE_X’
95 error   758 |       (node::addon_register_func) (regfunc),                          \
95 error       |                                    ^~~~~~~
95 error ../src/linux/BluetoothSerialPort.cc:23:1: note: in expansion of macro ‘NODE_MODULE’
95 error    23 | NODE_MODULE(BluetoothSerialPort, InitAll)
95 error       | ^~~~~~~~~~~
95 error make: *** [BluetoothSerialPort.target.mk:117: Release/obj.target/BluetoothSerialPort/src/linux/BluetoothSerialPort.o] Fehler 1
95 error gyp ERR! build error
95 error gyp ERR! stack Error: `make` failed with exit code: 2
95 error gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
95 error gyp ERR! stack     at ChildProcess.emit (events.js:400:28)
95 error gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
95 error gyp ERR! System Linux 5.11.0-37-generic
95 error gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "build"
95 error gyp ERR! cwd /tmp/obd2-test/node_modules/bluetooth-serial-port
95 error gyp ERR! node -v v14.17.5
95 error gyp ERR! node-gyp -v v7.1.2
95 error gyp ERR! not ok
96 verbose exit 1
```

**Comment by szaboge** [comment]
I found a solution for that: #37 
Works with node 12.

**Comment by EricSmekens** [comment]
Fixed by @szaboge in #37
